### PR TITLE
add patches for cloud-init CVE-2022-2084, CVE-2023-1786

### PR DESCRIPTION
**Pull request by rmhsawyer**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

add patches for cloud-init CVE-2022-2084, CVE-2023-1786

###### Change Log  

- add [upstream patch](https://github.com/canonical/cloud-init/commit/4d467b14363d800b2185b89790d57871f11ea88c#diff-5978682d3313395486da4847761e13e175e6303251fe7ddf81552db99779760eR459) for cloud-init CVE-2022-2084
- modify CVE-2023-1786.patch as it causes issue on azure VM deployment

###### Does this affect the toolchain?  

NO

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-1786
- https://nvd.nist.gov/vuln/detail/CVE-2022-2084

###### Test Methodology

- Pipeline build id: [386924](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=386924&view=results)
- Local package and image build
- Azure VM deployment

### CODE REVIEW

File: SPECS/cloud-init/cloud-init.spec [hunk 1]
```diff
@@ -163,8 +164,8 @@ rm -rf %{buildroot}
 %config(noreplace) %{_sysconfdir}/cloud/cloud.cfg.d/10-azure-kvp.cfg
 
 %changelog
-* Thu Jun 15 2023 Minghe Ren <mingheren@microsoft.com> - 21.4-3
-- Add patch for CVE-2023-1786
+* Tue Mar 22 2023 Minghe Ren <mingheren@microsoft.com> - 21.4-3
```
Comment (christopherco) [inline SPECS/cloud-init/cloud-init.spec:167]: Date

Comment (christopherco) [inline SPECS/cloud-init/cloud-init.spec:167]: Should this update be a release number increment to `-4`?

**Review by christopherco — CHANGES_REQUESTED** [review]

File: SPECS/cloud-init/cloud-init.spec (hunk 1 above)
Comment (rmhsawyer) [reply SPECS/cloud-init/cloud-init.spec:167]: I am thinking if I could do two patch change together as the previous PR for CVE-2023-1786 just merged but increment it makes it clearer.

Comment (rmhsawyer) [reply SPECS/cloud-init/cloud-init.spec]: Changed as suggested

**Review by christopherco — APPROVED** [review]
LGTM

Buddy Build passes:
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=389768&view=results
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=389767&view=results

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**